src/components/shared/FilterModal.tsx

```diff
@@ -0,0 +1,96 @@
+import classes from './FilterModal.module.scss';
+
+import { useContext, useState } from 'react';
+import { CategoriesContext } from '../../data/contexts/CategoriesContext';
+import { ICategoriesContext } from '../../data/contexts/CategoriesContext';
+import { Category } from '../../data/interfaces';
+import ModalPopup, { ButtonType } from '../generic/ModalPopup';
+import { deepEqual } from '../../utils/comparison';
+import Checkbox from '../input/checkbox/Checkbox';
+
+interface FilterModalProps {
+    onClose: () => void;
+}
+
+const FilterModal = ({ onClose }: FilterModalProps) => {
+    const { categories, selectedCategories, setSelectedCategories } =
+        useContext<ICategoriesContext>(CategoriesContext);
+
+    const [temporarilySelectedCategories, setTemporarilySelectedCategories] =
+        useState(selectedCategories);
+
+    const disabledPrimaryButton = deepEqual(
+        temporarilySelectedCategories,
+        selectedCategories
+    );
+
```

Comment:
Take a look across the project and see where else you use `deepEquel`. Wrap them all with `useMemo`